### PR TITLE
Added date-matching support for list_considering

### PR DESCRIPTION
**Pull request by Michael-Girma**
Modified list_considering's call to list to support date based matching for #7296 
---
- Passing date-based matching args to list function
- Created a test function that utilizes rlist to test list_considering's support

### CODE REVIEW

File: src/borg/manifest.py [hunk 1]
```diff
@@ -166,6 +166,10 @@ def list_considering(self, args):
             match=args.match_archives,
             first=args.first,
             last=args.last,
+            older=args.older,
+            newer=args.newer,
+            oldest=args.oldest,
+            newest=args.newest,
```
Comment (ThomasWaldmann) [inline src/borg/manifest.py:172]: did you check all callers of `list_considering` whether they set up these attributes of `args` (== whether they support the corresponding cmd line options in their arg parser)?

Comment (ThomasWaldmann) [reply src/borg/manifest.py:172]: (if the option parser for these options is not added, you'ld get an `AttributeError` here)

**Review by ThomasWaldmann — COMMENTED** [review]
Looks good, just 1 question.

File: src/borg/manifest.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/manifest.py:172]: maybe a less fragile way would be to use `x=getattr(args, "x", correctdefault)`.

i also missed to do that for first and last.

Comment (Michael-Girma) [reply src/borg/manifest.py:172]: Yeah, that seems like a better way to do it.

**Comment by ThomasWaldmann** [comment]
btw, if you mention "fixes #7296" in a commit comment or PR title, it will automatically close that issue if this is merged.

**Review by ThomasWaldmann — APPROVED** [review]
lgtm.

**Comment by ThomasWaldmann** [comment]
Thanks!

The linter (black) fail was strange btw:
- seems unrelated to your changes
- uses same black version on github as we use locally
- other PRs do not have this linter fail

**Pull request merged**